backend/capellacollab/core/authentication/api_key_cookie.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-FileCopyrightText: Copyright DB InfraGO AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import typing as t
+
+import fastapi
+import jwt
+from fastapi import security
+from jwt import exceptions as jwt_exceptions
+
+from capellacollab.config import config
+
+from . import exceptions, flow
+
+log = logging.getLogger(__name__)
+
+
+class JWTConfigBorg:
+    _shared_state: dict[str, str] = {}
+
+    def __init__(self) -> None:
+        self.__dict__ = self._shared_state
+
+        if not hasattr(self, "_jwks_client"):
+            self.jwks_client = jwt.PyJWKClient(
+                uri=flow.get_auth_endpoints()["jwks_uri"]
+            )
+
+        if not hasattr(self, "_supported_signing_algorithms"):
+            self.supported_signing_algorithms = (
+                flow.get_supported_signing_algorithms()
+            )
```

Comment:
There has been a change, the `JWTConfigBorg` now uses the oidc provider config class, so unfortunately it is no longer possible to use the `__new__` method here, since the provider config is passed as a parameter. I think it should be enough to transform this so that we use the `__init__` instead of `__new__`, but either you adjust this during the week, or I'll have a look at it next week.